### ISSUE TITLE
Iterating over a map by @each ... in ... breaks true.

### ISSUE THREAD

**Issue by hinok** [oddbird/true#74]
**Edit 2**
Could you clarify if a file that is tested, should it contain only SASS function and mixin declarations?

---

**Edit**
Small update, the same error appears for that scss

``` scss
$my-class: '.asd';

#{$my-class} {
  display: inline-block;
}
```

---

Hey,
I'm digging into `true` and after a few tests I've found a problem. It's not documented but it looks that I can't use `@each` in scss source file. When I remove `@each ... in ...`, everything works.

Source

``` scss
@mixin foo($name, $size) {
  .foo--#{$name} {
    width: $size;
  }
}

$sizes: (
  'aaa': 100px,
  'bbb': 200px
);

/* If I remove this line, it works */ @each $name, $size in $sizes { 
/* If I remove this line, it works */   @include foo($name, $size);   
/* If I remove this line, it works */ }                                                   
```

Test

``` scss
@import 'true';
@import './../src/foo';

@include test-module('@mixin foo') {
  @include test('should create a modifier class with specified width') {
    @include assert('...') {
      @include output {
        @include foo('aaa', 100px);
      }

      @include expect {
        .foo--aaa {
          width: 100px;
        }
      }
    }
  }
}
```

Full error

``` shell
> mocha tests/index.js --reporter spec --bail

my-package/node_modules/sass-true/lib/main.js:114
  throw parseError('Unexpected rule type "' + rule.type + '"', 'module header', rule.position);
  ^

Error: Line 1, column 1: Unexpected rule type "rule"; looking for module header
    at parseError (my-package/node_modules/sass-true/lib/main.js:89:10)
    at parseModule (my-package/node_modules/sass-true/lib/main.js:114:9)
    at my-package/node_modules/sass-true/lib/main.js:79:15
    at Function._.each._.forEach (my-package/node_modules/underscore/underscore.js:153:9)
    at module.exports.parse (my-package/node_modules/sass-true/lib/main.js:78:5)
    at Object.module.exports.runSass (my-package/node_modules/sass-true/lib/main.js:35:17)
    at Object.<anonymous> (my-package/tests/index.js:5:10)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:20:19)
    at my-package/node_modules/mocha/lib/mocha.js:220:27
    at Array.forEach (native)
    at Mocha.loadFiles (my-package/node_modules/mocha/lib/mocha.js:217:14)
    at Mocha.run (my-package/node_modules/mocha/lib/mocha.js:485:10)
    at Object.<anonymous> (my-package/node_modules/mocha/bin/_mocha:457:18)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:148:18)
    at node.js:405:3

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "test:scss"
npm ERR! node v5.11.0
npm ERR! npm  v3.9.6
npm ERR! code ELIFECYCLE
npm ERR! my-package@0.0.1 test:scss: `mocha tests/index.js --reporter spec --bail`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the my-package@0.0.1 test:scss script 'mocha tests/index.js --reporter spec --bail'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the my-package package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     mocha tests/index.js --reporter spec --bail
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs my-package
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls my-package
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     my-package/npm-debug.log
```

true 2.1.3
node 5.11

**Comment by mirisuzanne** [comment]
@hinok Yes, the error comes from having unexpected css in the output. Our parser isn't handling that well.

@carljm I think we discussed something similar around loud comments, but I don't remember all the implications. Would it be hard for us to ignore CSS that falls outside our expected true-output syntax — or should we document that source files should not have output?

**Comment by mirisuzanne** [comment]
Resolved by #97
